testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/realm/RealmTest.java

```diff
@@ -1220,4 +1220,27 @@ public void testSetEmptyAttributeValues() {
             adminClient.realm(realmName).remove();
         }
     }
+
+    @Test
+    public void testRefreshTokenPersistence() {
```

Comment:
This test is not correct as it does not test at all the scenario from the description.

I've tried to copy/paste the test to my fork and it passed even without your change in the `OIDCIdentityProvider` . Which is because the test you added just adds the regular refresh token (without any involvement from identity brokering...)

The correct test is the one, which really tests the scenario you are fixing. So likely should fail without your change and should pass after your change. AFAIK the scenarios related to OIDC identity brokering and token exchange are in `KcOidcBrokerTokenExchangeTest`